extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -286,6 +287,24 @@ export class AmpAnalytics extends AMP.BaseElement {
     return Promise.all(promises);
   }
 
+ /**
+  * Gets the resource ID of the amp-ad element containing this AmpAnalytics
+  * instance.
+  * If there is no containing amp-ad tag, then the resource ID of this
+  * amp-analytics tag is used instead.
+  * @return {string}
+  */
+  getAmpAdResourceId() {
+    try {
+      const frame = getParentWindowFrameElement(this.element, this.win.top);
+      return frame.parentElement.getResourceId();
+    } catch (e) {
+      this.user().error(TAG, 'No friendly parent amp-ad element was found' +
```

Comment:
This should never happen right?  Perhaps we shouldn't try to recover...

Reply:
What if the amp-analytics tag (using iframe transport) is placed directly within the publisher page? We're not addressing that use case at this time, but someone could certainly attempt it...

Reply:
Ok, I changed it to not recover, but added a comment that non-A4A may be enabled in the future.
We can design that later, if any customer wants it, but I believe that using the resourceId of the AmpAnalytics instance itself will still be a unique number so collisions would be avoided.
I can add doc saying that transport: iframe is A4A-only - great idea.

Reply:
Sounds good. Can you create an issue and add a TODO here with the issue number?

Reply:
Done